backend/controllers/auth_controller.go

```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/ZPI-2024-25/KubernetesAccessManager/auth"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+func CheckLoginStatus(w http.ResponseWriter, r *http.Request)  {
+	token, err := auth.GetJWTTokenFromHeader(r)
+	isValid, claims := auth.IsTokenValid(token)
+
+	if err != nil || !isValid {
+		writeJSONResponse(w, http.StatusUnauthorized, models.ModelError{
+			Message: "Unauthorized",
+			Code: http.StatusUnauthorized,
+		})
+		return
+	}
+	rolemap, err := auth.GetRoleMapInstance()
+	if err != nil {
+		writeJSONResponse(w, http.StatusInternalServerError, &models.ModelError{
+			Message: "Internal Server Error",
+			Code: http.StatusInternalServerError,
+		})
+	}
+	status, errM := getLoginStatus(claims, rolemap)
+	if errM != nil {
+		writeJSONResponse(w, int(errM.Code), err)
+		return
+	}
+	writeJSONResponse(w, http.StatusOK, status)
+}
+
+func getLoginStatus(claims *jwt.MapClaims, rolemap *auth.RoleMapRepository) (*models.UserStatus, *models.ModelError) {
+	roles, err := auth.ExtractRoles(claims)
+	if err != nil {
+		return nil, &models.ModelError{
+			Message: "Roles not found in bearer token",
+			Code: http.StatusInternalServerError,
+		}
+	}
+	access := rolemap.GetAllPermissions(roles)
+	auth.PrunePermissions(access)
+	permissions := shortenPermissions(access)
+	exp, preferredUsername, email := auth.ExtractUserStatus(claims)
+
+	return &models.UserStatus{
+		Permissions: permissions,
+		User: &models.UserStatusUser{
+			PreferredUsername: preferredUsername,
+			Email: email,
+			Exp: exp,
+		},
+	}, nil
+}
+
+func shortenPermissions(pmatrix map[string]map[string]map[models.OperationType]struct{}) map[string]map[string][]string {
+	res := make(map[string]map[string][]string)
+
+	for k, v := range pmatrix {
+		res[k] = make(map[string][]string)
+		for k2, v2 := range v {
+			res[k][k2] = make([]string, len(v2))
+			i := 0
+			for k3 := range v2 {
+				res[k][k2][i] = shorterP(k3)
+				i++
+			}
+		}
+	}
+	return res
+}
+
+func shorterP(p models.OperationType) string {
```

Comment:
In design of this function I've decided to be as direct as possible. This function takes in full Permission and turns it into one letter P, thus shorterP

Reply:
Thinking about it I could move this function to models package and make it a method + name it "p"

Reply:
Let @Ninja00Shadow comment further, but to my mind quite a few of the names that occur in your code are relatively nondescript and it's easy to get lost in the algorithmic functions of what's going on. If there was an option, it would be worth changing this and other names to more descriptive ones (but without exaggeration), because shorterP if it occurs out of context says absolutely nothing